pkg/operator/ceph/controller/predicate.go

```diff
@@ -199,35 +224,79 @@ func WatchPredicateForNonCRDObject(owner runtime.Object, scheme *runtime.Scheme)
 		CreateFunc: func(e event.CreateEvent) bool {
 			return false
 		},
+
 		DeleteFunc: func(e event.DeleteEvent) bool {
 			match, object, err := ownerMatcher.Match(e.Object)
 			if err != nil {
 				logger.Errorf("failed to check if object kind %q matched. %v", e.Object.GetObjectKind(), err)
 			}
+			objectName := object.GetName()
 			if match {
-				logger.Debugf("object %q matched on delete", object.GetName())
+				// If the resource is a CM, we might want to ignore it since some of them are ephemeral
+				isCMToIgnoreOnDelete := isCMToIgnoreOnDelete(e.Object)
+				if isCMToIgnoreOnDelete {
+					return false
+				}
+
+				// If the resource is a canary deployment we don't reconcile because it's ephemeral
+				isCanary := isCanary(e.Object)
+				if isCanary {
+					return false
+				}
+
+				logger.Infof("object %q matched on delete, reconciling", objectName)
 				return true
 			}
 
-			logger.Debugf("object %q did not match on delete", object.GetName())
+			logger.Debugf("object %q did not match on delete", objectName)
 			return false
 		},
+
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			match, object, err := ownerMatcher.Match(e.ObjectNew)
 			if err != nil {
 				logger.Errorf("failed to check if object matched. %v", err)
 			}
+			objectName := object.GetName()
 			if match {
-				logger.Debugf("object %q matched on update", object.GetName())
-				objectChanged, err := objectChanged(e.ObjectOld, e.ObjectNew)
+				logger.Debugf("object %q matched on update", objectName)
```

Comment:
There are so many cases we want to skip during the update, is it even worth triggering the reconcile on update? What important scenarios still get handled during the update? If they're all corner cases, maybe it's not worth it.

Reply:
I've been thinking about this too and perhaps I'm trying to reconcile as much as possible to enhance the robustness of the Operator but there are still Secret and CM we might want to reconcile on update but I'll dig again.

Reply:
Ok found a good example, we need to reconcile on the CM rook-config-override.